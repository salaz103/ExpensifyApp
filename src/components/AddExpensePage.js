import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {addExpense} from '../actions/expenses';

const AddExpensePage=(props)=>(
    <div>
        <h1>Add Expense</h1>
        <ExpenseForm
            onSubmit={(expense)=>{
                props.dispatch(addExpense(expense));
                //ESTO USA EL BROWSER ROUTING, PARA REDIRECCIONAR A LA PAGINA DEL "DASHBOARD"
                //DESPUES DE HABER DESPACHADO LA ACCION A LA STORE
                props.history.push('/');
            }}
        />
    </div>
);

export default connect()(AddExpensePage);