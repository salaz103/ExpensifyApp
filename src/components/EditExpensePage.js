import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';





export class EditExpensePage extends React.Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                {/* Editing the expense with ID of {props.match.params.id} */}
                <div className="content-container">
                    <ExpenseForm
                        //ESTE "EXPENSE" ES EL QUE ESTAMOS ENVIANDO PARA QUE LLENE LOS CAMPOS EN EL "FORM"    
                        expense={this.props.expense}
                        //ESTE "EXPENSE" ES EL QUE ESTA SUBIENDO, YA SEA CON INFORMACION NUEVA O SIN CAMBIOS 
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>
                </div>
            </div>
        );
    }

}

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))

});


const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id;
        })
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);