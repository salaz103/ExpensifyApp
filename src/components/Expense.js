import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

            //AQUI SE ESTA HACIENDO UN SPREAD (...) DEL OBJETO, PERO TAMBIEN SE PUEDE LLAMAR A "DISPATCH"
const ExpenseListItem= ({description,amount,createdAt,id})=>(
    <Link className="list-item" to={`/edit/${id}`}>
    <div>
    <h3 className="list-item__title">{description}</h3>
    <span className="list-item__subtitle">Created At: {moment(createdAt).format('MMMM Do, YYYY')}</span>
    </div>
    <h3 className="list-item__data">Amount: {numeral(amount/100).format('$0,0.00')}</h3>
    </Link>
);

export default ExpenseListItem;