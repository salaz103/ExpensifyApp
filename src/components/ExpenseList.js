import React from 'react';
import {connect} from 'react-redux';


const ExpenseList= (props)=>(
    <div>
        <h1>Expense List</h1>
        {props.expenses.length}
    </div>
);

const mapStatetoProps= (state)=>{
    return{
        expenses: state.expenses,
        filters: state.filters
    };
};

const ConnectedExpenseList= connect(mapStatetoProps)(ExpenseList);

export default ConnectedExpenseList;