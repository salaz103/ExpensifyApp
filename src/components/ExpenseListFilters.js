import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';
import { DateRangePicker } from 'react-dates';
import 'react-dates/initialize';
import { sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({
            calendarFocused
        }));
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    onSortChange = (e) => {
        e.target.value === "date" ? this.props.sortByDate() : null;
        e.target.value === "amount" ? this.props.sortByAmount() : null;
    }

    render() {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange} />

                <select
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>

                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    startDateId="addExpense_start_date"
                    endDate={this.props.filters.endDate}
                    endDateId="addExpense_end_date"
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => { false }}
                />

            </div>

        );
    }

}


const mapDispatchToProps=(dispatch)=>({
    setTextFilter: (text)=> dispatch(setTextFilter(text)),
    sortByDate: ()=> dispatch(sortByDate()),
    sortByAmount: ()=>dispatch(sortByAmount()),
    setStartDate: (startDate)=>dispatch(setStartDate(startDate)),
    setEndDate: (endDate)=>dispatch(setEndDate(endDate))
});

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);