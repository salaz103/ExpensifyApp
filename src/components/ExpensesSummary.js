import React from 'react';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';
import selectTotalExpenses from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';



export const ExpensesSummary = (props) => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{props.expenseCount}</span>
                {props.expenseCount == 0 || props.expenseCount > 1 ? ' expenses ' : ' expense '}
            totalling <span>{numeral(props.expenseTotal / 100).format('$0,0.00')}</span>
            </h1>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
    </div>
);


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectTotalExpenses(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);