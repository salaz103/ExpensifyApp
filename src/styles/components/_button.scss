.button{
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

//THIS IS A MODIFIER FOR LINK, THATS WHY WE USED --
// IT IS LIKE IF WE USED @EXTEND 
.button--link{
    background: none;
}

.button--secondary{
    background: #888888;
}