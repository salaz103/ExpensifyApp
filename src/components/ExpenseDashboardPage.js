import React from 'react';
import ExpenseList from './ExpenseList';


const ExpenseDashboardPage=()=>(
    <ExpenseList/>
);


export default ExpenseDashboardPage;