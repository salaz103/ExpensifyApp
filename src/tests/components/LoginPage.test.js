import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';


test('Should render LoginPage',()=>{
    const wrapper = shallow(<LoginPage startLogin={()=>{ }}/>);
    expect(wrapper).toMatchSnapshot();
});


test('Should call startLogin on button click',()=>{
    //WE WILL USE A SPY TO SIMULATE IF THE LOGIN BUTTON WAS SUCCESFULLY CALLLED
    const startLogin= jest.fn();
    const wrapper = shallow(<LoginPage startLogin= {startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});