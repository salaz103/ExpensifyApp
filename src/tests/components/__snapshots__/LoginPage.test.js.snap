// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify App
    </h1>
    <p>
      Manage your expenses, save money.
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
